neqo-transport/src/tracking.rs

```diff
@@ -451,29 +460,77 @@ impl AckTracker {
     }
 }
 
-impl Default for AckTracker {
-    fn default() -> Self {
-        Self {
-            spaces: [
-                RecvdPackets::new(PNSpace::Initial),
-                RecvdPackets::new(PNSpace::Handshake),
-                RecvdPackets::new(PNSpace::ApplicationData),
-            ],
-        }
+impl ::std::fmt::Display for RecvdPackets {
+    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        write!(f, "Recvd-{}", self.space)
     }
 }
 
-impl Index<PNSpace> for AckTracker {
-    type Output = RecvdPackets;
+#[derive(Debug)]
+pub struct AckTracker {
+    /// This stores information about received packets in *reverse* order
+    /// by spaces.  Why reverse?  Because we ultimately only want to keep
+    /// `ApplicationData` and this allows us to drop other spaces easily.
+    spaces: SmallVec<[RecvdPackets; 3]>,
```

Comment:
This might be getting way too clever, but I had the thought of declaring the smallvec of length 1, so that space for handshake and initial entries would be dynamically allocated but then the alloc freed once they were removed...

Reply:
Oh, I like that.  It's clever, but it's not overly so.